src/main/java/org/folio/circulation/services/EventPublisher.java

```diff
@@ -130,9 +131,17 @@ public CompletableFuture<Result<CheckInContext>> publishItemCheckedInEvents(
       .thenCombineAsync(loanRepository.findLastLoanForItem(checkInContext.getItem().getItemId()), (userResult, lastLoan) -> {
         if (nonNull(lastLoan.value())) {
           return userRepository.getUser(lastLoan.value().getUserId())
-            .thenApply(userFromLastLoan -> Result.succeeded(pubSubPublishingService.publishEvent(LOG_RECORD.name(),
-              mapToCheckInLogEventContent(checkInContext, userResult.value(),
-                checkInContext.isInHouseUse() ? null : userFromLastLoan.value()))));
+            .thenApply(userFromLastLoan -> {
+              try {
+                return Result.succeeded(pubSubPublishingService.publishEvent(LOG_RECORD.name(),
+                  mapToCheckInLogEventContent(checkInContext, userResult.value(),
+                    (checkInContext.isInHouseUse() || !loanRepository.hasOpenLoan(checkInContext.
+                      getItem().getItemId()).get().value() || checkInContext.getRequestQueue().getRequests().isEmpty())
```

Comment:
Calling `CompletableFuture#get` will block the thread, please avoid it.
Calling `Result#value` will return null if result is failed, beware of NPEs. In general, the use of this method is discouraged. Consider using built-in methods, such as `Result#map` or `Result#next`, for result composition instead.